cubeviz/tools/smoothing.py

```diff
@@ -0,0 +1,334 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import sys
+
+import astropy.units as u
+from astropy import convolution
+
+from glue.core import Data, Subset
+from glue.core.component import Component
+from glue.core import message as msg
+from glue.core.coordinates import coordinates_from_header, WCSCoordinates
+from glue.core.exceptions import IncompatibleAttribute
+
+from spectral_cube import SpectralCube, masks, BooleanArrayMask
+
+__all__ = ["smooth", "data_to_cube", "cube_to_glueData",
+    "get_kernel_registry", "print_kernel_types", "list_kernel_types"]
+
+def _get_glue_mask(data, component_id):
+    mask = None
+    if isinstance(data, Subset):
+        try:
+            mask = data.to_mask()
+            return mask
+        except IncompatibleAttribute:
+            pass
+
+    d = data[component_id]
+    mask = np.empty_like(d)
+    mask.fill(True)
+    mask = mask.astype(bool)
+
+    return mask
+
+def _get_glue_wcs(data):
+    coords = data.coords
+    if isinstance(coords, WCSCoordinates):
+        return coords.wcs
+    raise Exception("WCS information was not provided.")
+
+def cube_to_glueData(cube, output_label="SmoothedCube",
+    output_component_id="Cube", output_as_component=False,
+    original_data=None):
+    if output_as_component:
+        original_data.add_component(cube._data, output_component_id)
+        original_data.broadcast(
+            msg.DataAddComponentMessage(
+                original_data,
+                output_component_id
+            )
+        )
```

Comment:
I think the DataAddComponentMesssage should be automatically emitted when calling ``add_component`` above - if not, that's a bug in glue.